Bug: Fix ESLint warning - unused bobId variable in combat.test.ts

## Description
There is an unused variable warning in the combat test file.

## Location
- File: \`src/lib/game-state/__tests__/combat.test.ts\`
- Line: 746

## Issue
\`\`\`
  746:20  warning  'bobId' is assigned a value but never used  @typescript-eslint/no-unused-vars
\`\`\`

## Solution
Remove the unused \`bobId\` variable from the destructuring on line 746:
\`\`\`typescript
// Change from:
const { state, bobId } = setupGameWithCreatures([], []);

// To:
const { state } = setupGameWithCreatures([], []);
\`\`\`

## Acceptance Criteria
- [ ] ESLint warning is resolved
- [ ] All tests still pass

--- a/src/lib/game-state/__tests__/combat.test.ts
+++ b/src/lib/game-state/__tests__/combat.test.ts
@@ -743,7 +743,7 @@ describe('Combat System - Edge Cases', () => {
   });
 
   it('should handle no attackers declared', () => {
-    const { state, bobId } = setupGameWithCreatures([], []);
+    const { state } = setupGameWithCreatures([], []);
 
     state.turn.currentPhase = Phase.DECLARE_ATTACKERS;
 

